test/pytest/service-desktop/test_contacts.py

```diff
@@ -1,146 +1,357 @@
+
+    third_contact_record = {
+        "address": "6 Czeczota St.\n02600 Warsaw",
+        "altName": "Williams",
+        "email": "george.williams@mudita.com",
+        "blocked": False,
+        "favourite": False,
+        "ice": False,
+        "numbers": [
+            "323456789"
+        ],
+        "speedDial": "3",
+        "priName": "George",
+        "note": "Some note"
+    }
+
+    result, third_contact_id = contacts_tester.add_contact(third_contact_record)
+    assert result, "Failed to add contact!"
+    result, retrieved_contact_record = contacts_tester.get_contact_by_id(third_contact_id)
+    assert result, "Failed to get contact by ID!"
+    del retrieved_contact_record["id"]
+    assert third_contact_record == retrieved_contact_record
+    current_number_of_contact_records += 1
+
+    fourth_contact_record = {
+        "address": "6 Czeczota St.\n02600 Warsaw",
+        "altName": "Brown",
+        "email": "olivia.brown@mudita.com",
+        "blocked": False,
+        "favourite": False,
+        "ice": False,
+        "numbers": [
+            "423456789"
+        ],
+        "speedDial": "4",
+        "priName": "Olivia",
+        "note": "Some note"
+    }
+
+    result, fourth_contact_id = contacts_tester.add_contact(fourth_contact_record)
+    assert result, "Failed to add contact!"
+    result, retrieved_contact_record = contacts_tester.get_contact_by_id(fourth_contact_id)
+    assert result, "Failed to get contact by ID!"
+    del retrieved_contact_record["id"]
+    assert fourth_contact_record == retrieved_contact_record
+    current_number_of_contact_records += 1
+
+    # Check contacts counter
+    result, received_contact_records_count = contacts_tester.get_contacts_count()
+    assert result, "Failed to get contacts count!"
+    assert current_number_of_contact_records == received_contact_records_count
+
+    # Get all previously added contacts
+    result, returned_contacts, total_count = contacts_tester.get_contacts_with_offset_and_limit(
+        initial_number_of_contact_records, current_number_of_contact_records - initial_number_of_contact_records)
+    assert result, "Failed to get contacts with offset and limit!"
+
+    for record in returned_contacts:
+        print(record)
+        del record["id"]
+    # Returned contacts are sorted by altName
+    assert returned_contacts[0] == fourth_contact_record
+    assert returned_contacts[1] == second_contact_record
+    assert returned_contacts[2] == first_contact_record
+    assert returned_contacts[3] == third_contact_record
+    assert current_number_of_contact_records == total_count
+
+    # Get previously added contacts without 2 of them (test offset)
+    result, returned_contacts, total_count = contacts_tester.get_contacts_with_offset_and_limit(
+        initial_number_of_contact_records + 2, current_number_of_contact_records - initial_number_of_contact_records)
+    assert result, "Failed to get contacts with offset and limit!"
+
```

Comment:
You are reusing this loop and asserts so many times that you might as well write a method for it and use it in the test case. This way, this code will be much more readable and you will be able to change the number of contacts if necessary instead of hard coding it in the scenario.

Reply:
Actually, the lines you are pointing are not repeated, they just look similar in two other places.
But of course, I see some space for improvements and optimization here in upcoming iterations.